firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift

```diff
@@ -0,0 +1,902 @@
+    }
+
+    private let connectionDetailArrow: UIImageView = .build { image in
+        image.image = UIImage(imageLiteralResourceName: StandardImageIdentifiers.Large.chevronLeft)
+            .withRenderingMode(.alwaysTemplate)
+            .imageFlippedForRightToLeftLayoutDirection()
+        image.transform = CGAffineTransform(rotationAngle: .pi)
+    }
+
+    private let connectionButton: UIButton = .build { button in }
+    private let connectionHorizontalLine: UIView = .build { _ in }
+
+    // MARK: Toggle View
+    private let toggleView: UIView = .build { view in
+        view.layer.cornerRadius = TPMenuUX.UX.viewCornerRadius
+        view.layer.maskedCorners = [.layerMinXMaxYCorner, .layerMaxXMaxYCorner]
+        view.backgroundColor = .clear
+        view.translatesAutoresizingMaskIntoConstraints = false
+    }
+    private let toggleLabelsContainer: UIStackView = .build { stack in
+        stack.backgroundColor = .clear
+        stack.distribution = .fill
+        stack.alignment = .leading
+        stack.axis = .vertical
+        stack.spacing = TPMenuUX.UX.headerLabelDistance
+    }
+
+    private let toggleLabel: UILabel = .build { label in
+        label.font = FXFontStyles.Regular.body.scaledFont()
+        label.numberOfLines = 0
+        label.adjustsFontForContentSizeCategory = true
+    }
+
+    private let toggleSwitch: UISwitch = .build { toggleSwitch in
+        toggleSwitch.isEnabled = true
+    }
+
+    private let toggleStatusLabel: UILabel = .build { label in
+        label.font = FXFontStyles.Regular.caption1.scaledFont()
+        label.numberOfLines = 0
+        label.adjustsFontForContentSizeCategory = true
+    }
+
+    // MARK: Clear Cookies View
+    private lazy var clearCookiesButton: TrackingProtectionButton = .build { button in
+        button.titleLabel?.textAlignment = .left
+        button.titleLabel?.numberOfLines = 0
+        button.layer.cornerRadius = TPMenuUX.UX.clearDataButtonCornerRadius
+        button.layer.borderWidth = TPMenuUX.UX.clearDataButtonBorderWidth
+        button.addTarget(self, action: #selector(self.didTapClearCookiesAndSiteData), for: .touchUpInside)
+    }
+
+    // MARK: Protection setting view
+    private lazy var settingsLinkButton: LinkButton = .build { button in
+        button.titleLabel?.textAlignment = .left
+        button.titleLabel?.numberOfLines = 0
+        button.addTarget(self, action: #selector(self.protectionSettingsTapped), for: .touchUpInside)
+    }
+
+    private var constraints = [NSLayoutConstraint]()
+
+    // MARK: - Variables
+
+    private var viewModel: TrackingProtectionModel
+    private var hasSetPointOrigin = false
+    private var pointOrigin: CGPoint?
+    var asPopover = false
+
+    private var toggleContainerShouldBeHidden: Bool {
+        return !viewModel.globalETPIsEnabled
+    }
+
+    private var protectionViewTopConstraint: NSLayoutConstraint?
+
+    // MARK: - View lifecycle
+
+    init(viewModel: TrackingProtectionModel,
```

Comment:
can we add tests for this class since its a new file?

Reply:
For TrackingProtectionModel, or?

Reply:
for `TrackingProtectionViewController`

Reply:
there are some tests added in the `TrackingProtectionStateTests` and more will come after this PR gets merged and we can connect the `TrackingProtectionViewController` with the redux actions and flows

Reply:
I was thinking at least memory leak test as a minimum, but this can be done in a separate PR so not blocking this one!